cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/queue/SimpleQueueSpecificationGenerator.java

```diff
@@ -56,15 +57,17 @@
   /**
    * Account Name under which the stream names to generated.
    */
-  private final Id.Application appId;
+  private final ApplicationId appId;
+  private final Id.Application idApplication;
 
   /**
    * Constructor that takes an appId.
    *
    * @param appId under which the stream is represented.
    */
-  public SimpleQueueSpecificationGenerator(Id.Application appId) {
+  public SimpleQueueSpecificationGenerator(ApplicationId appId) {
     this.appId = appId;
+    idApplication = appId.toId();
```

Comment:
why is this needed? instead just update ``generateQueueSpecification`` to accept ``ApplicationId``?